src/ansys/dpf/post/tools/streamlines.py

```diff
@@ -0,0 +1,76 @@
+"""Module containing the helpers for streamlines.
+
+Streamlines
+-----------
+
+"""
+from typing import List, Union
+
+from ansys.dpf.core.helpers import streamlines as core_streamlines
+from ansys.dpf.core.plotter import DpfPlotter
+
+from ansys.dpf import post
+
+
+def plot_streamlines(
+    dataframe: post.DataFrame,
+    sources: List[dict],
+    mesh: Union[post.Mesh, None] = None,
+    plot_mesh: bool = True,
+    mesh_opacity: float = 0.3,
+    plot_contour: bool = True,
+    contour_opacity: float = 0.3,
+    **kwargs,
+):
+    """Plot streamlines based on a vector field DataFrame.
+
+    Parameters
+    ----------
+    dataframe:
+        A `post.DataFrame` object containing a vector field.
+    sources:
+        A list of dictionaries defining spherical point sources for the streamlines.
+        Expected keywords are "center", "radius", and "line_thickness".
+    mesh:
+        A `post.Mesh` object to use to compute the streamlines instead of the mesh associated with
+        the vector field in the DataFrame.
+    plot_contour:
+        Whether to plot the field's norm contour along with the streamlines.
+    contour_opacity:
+        Opacity to use for the field contour in case "plot_contour=True".
+    plot_mesh:
+        Whether to plot the mesh along the streamlines in case "plot_contour=False".
+    mesh_opacity:
+        Opacity to use for the mesh in case "plot_contour=False" and "plot_mesh=True".
+    **kwargs:
+
+    """
+    # Select data to work with
+    if mesh:
+        meshed_region = mesh._meshed_region
+    else:
+        meshed_region = dataframe._fc[0].meshed_region
```

Comment:
we will eventually need a public method in the dataframe to get the mesh, no?

Reply:
@cbellot000 sure, but again it will need a label logic to select which field's meshed_region is of interest, or else I guess return a `MeshesContainer` as a `Meshes`